app/cdash/xml_handlers/upload_handler.php

```diff
@@ -110,51 +111,20 @@ public function startElement($parser, $name, $attributes)
             // when we call UpdateBuild() below.
             //
             // For end time, we use the start of the testing day.
-            // For start time, we use either the submit time (now) or
-            // one second before the start time of the *next* testing day
-            // (whichever is earlier).
+            // For start time, we use the end of the testing day.
             // Yes, this means the build finished before it began.
             //
             // This associates the build with the correct day if it is only
             // an upload.  Otherwise we defer to the values set by the
             // other handlers.
-
-            $db = Database::getInstance();
-            $row = $db->executePreparedSingleRow('
-                       SELECT nightlytime FROM project where id=?
-                   ', [intval($this->projectid)]);
-            $nightly_time = $row['nightlytime'];
-            $build_date =
+            $buildDate =
                 extract_date_from_buildstamp($this->Build->GetStamp());
+            list($beginningOfDay, $endOfDay) =
+                $this->Project->ComputeTestingDayBounds($buildDate);
```

Comment:
As of PHP 7.1, you do not need to use the `list()` or `array()` constructs.  I plan to make a style fixer PR to enforce this once I get my older PRs merged.

```suggestion
            [$beginningOfDay, $endOfDay] = $this->Project->ComputeTestingDayBounds($buildDate);
```